### PR TITLE
Add pipeline support

### PR DESCRIPTION
**Pull request by messense**

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -161,14 +162,175 @@ impl Client {
...
+    ///   # Your code here.
+    /// });
+    /// pipe.send();
+    /// ```
+    pub fn time<F>(&mut self, metric: &str, callable: F)
+        where F: Fn()
+    {
+        let start = clock_ticks::precise_time_ms();
+        callable();
+        let end = clock_ticks::precise_time_ms();
```
Comment (markstory) [inline src/client.rs:321]: I wonder if there is a way we could avoid duplicating this and some of the other method logic. Perhaps if the `Client` was wrapped in a decorator that provided a stubbed `send()` method? That might let `Pipeline` simply delegate to the pipeline 'client'.

File: src/client.rs [hunk 2]
```diff
@@ -161,14 +162,175 @@ impl Client {
...
+    }
+
+    /// Send data along the UDP socket.
+    pub fn send(&mut self) {
+        let mut _data = String::new();
+        if let Some(data) = self.stats.pop_front() {
+            _data = _data + &data;
+            while !self.stats.is_empty() {
+                let stat = self.stats.pop_front().unwrap();
+                if data.len() + stat.len() + 1 > 512 {
```
Comment (markstory) [inline src/client.rs]: Should the UDP packet size be a parameter, or struct property?

File: src/client.rs [hunk 3]
```diff
@@ -161,14 +162,175 @@ impl Client {
...
+
+impl Pipeline {
+    fn new(client: Client) -> Pipeline {
+        Pipeline {
+            client: client,
+            stats: VecDeque::new(),
+        }
+    }
+
+        /// Increment a metric by 1
```
Comment (markstory) [inline src/client.rs]: Indentation is of here.

File: README.md [hunk 4]
```diff
@@ -56,3 +56,26 @@ client.time("operation.duration", || {
 	// Do something expensive.
 });
 ```
+
+### Pipeline
+
+Multiple metrics can be sent to StatsD once using pipline:
```
Comment (markstory) [inline README.md]: Typo in pipeline.

**Comment by markstory** [comment]
Looking good!

File: src/client.rs (hunk 2 above)
Comment (messense) [inline src/client.rs]: Not sure how to expose the parameter to user.

File: src/client.rs (hunk 1 above)
Comment (messense) [inline src/client.rs:321]: I'll think about it.

File: README.md [hunk 5]
```diff
@@ -56,3 +56,29 @@ client.time("operation.duration", || {
...
+pipe.decr("some.counter");
+
+// Update a gauge
+pipe.gauge("some.value", 12.0);
+
+// Modify a counter by an arbitrary float.
+pipe.count("some.counter", 511.0);
+
+// Set max UDP packet size if you wish, default is 512
+pipe.set_max_udp_size(128);
```
Comment (markstory) [inline README.md:80]: :+1:

**Comment by markstory** [comment]
Thanks! Tagged and packaged into 0.5.0

**Comment by messense** [comment]
I think maybe we could make the pipeline API more like `redis-rs`:

``` rust
extern crate statsd;

use statsd::Client;

let client = Client::new("127.0.0.1:8125", "myapp").unwrap();

let mut pipe = statsd::pipeline();
pipe.incr("some.counter");
pipe.decr("some.counter");
pipe.send(&client);
```

**Comment by markstory** [comment]
@messense That sounds like a simpler API :+1:. Did you want to do that change or should I do it?

**Comment by messense** [comment]
@markstory I don't have much time now. You should do it. :-)

**Comment by markstory** [comment]
@messense Ok.

**Pull request merged**